plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegate.java

```diff
@@ -75,6 +75,7 @@ private AppEngineRuntimeInstallJob(IFacetedProject facetedProject) {
       super(Messages.getString("appengine.install.runtime.to.project", // $NON-NLS$
           facetedProject.getProject().getName()));
       this.facetedProject = facetedProject;
+      setRule(facetedProject.getProject());
```

Comment:
Um... wasn't talking about that exactly (I was wrong), but I think you got another point. The suspender will suspend the second `ConvertJob`, but it won't suspend the first `ConvertJob`. We have this comment in `AppEngineFacetInstallDelegate`:

```java
    installJob.schedule();
    // https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1155
    // The first ConvertJob has already been scheduled (which installs JSDT facet), and
    // this is to suspend the second ConvertJob temporarily.
    NonSystemJobSuspender.suspendFutureJobs();
  }
```
The first `ConvertJob` is the one that installs the JSDT facet (you'll see the comment in the code as well), and it does `commitChanges()`. (I don't think it tries to acquire a workspace lock, but I think it'll be blocked on the lock to modify project facets anyway.) The `AppEngineRuntimeInstallJob.waitUntilJsdtIsFixedFacet()` waits for the first `ConvertJob` to finish, but if we're holding a project lock, I think we'll get into a deadlock. (Not tested though.)

I may be wrong, however. Cannot be confident until we do some actual testing.